### DISCUSSION TITLE
Support ConPTY as an replacement to STDIO redirection in System.Diagnostics.Process

### DISCUSSION THREAD

**Discussion by wfjsw** [dotnet/runtime#96000] [Ideas]
In 2018 Windows released [Windows Pseudo Console (ConPTY)](https://devblogs.microsoft.com/commandline/windows-command-line-introducing-the-windows-pseudo-console-conpty/) and I am looking for ways to integrate it into some of my subprocesses.

I quickly find out that:
1. I cannot turn a `PROCESS_ALL_ACCESS` handle into a `Process` object, if I were to occupy the handle by [`CreateProcess`](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessa) on my own. The only way I can do it right now is to firstly convert the handle to a PID, then use the `GetProcessById` to create the `Process` object. In this process it involves [`OpenProcess`](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-openprocess) with `PROCESS_ALL_ACCESS` which in turn requires `SeDebugPrivilege`. This is bad and error-prone.
2. To use ConPTY I have to use `STARTUPINFOEX` and put something inside, instead of `STARTUPINFO`, but nevertheless even the `STARTUPINFO` is beyond my control. 

To implement it I end up cloning the entire `Process.Start()` flow to make the modification, though I don't really have much idea on how to make it better.

**Comment by alexrp** [comment]
There is a general trend of people needing more control over process creation. Related issues:

* https://github.com/dotnet/runtime/issues/94127
* https://github.com/dotnet/runtime/issues/71595
* https://github.com/dotnet/runtime/issues/71515
* https://github.com/dotnet/runtime/issues/13943
